### PR TITLE
Add adopters page and YAML config

### PR DESCRIPTION
**Pull request by dholbach**
For Flux v1 and Flux v2, driven by yaml files.

Fixes: #129 

~~Needs #176 to be landed first.~~

### CODE REVIEW

**Comment by dholbach** [comment]
Needs https://mikefarah.github.io/yq/ ...

**Comment by dholbach** [comment]
This is not very sophisticated yet, but see https://github.com/dholbach/website/blob/fix-129/adopters/flux-v2.yaml from which https://609019180b1d6800085b92f0--fluxcd.netlify.app/adopters/flux-v2/ is generated. It uses a modified form of https://www.docsy.dev/docs/adding-content/shortcodes/#card-panes which we get through #176 .

It'd be nice to simplify https://github.com/dholbach/website/blob/fix-129/Makefile#L49-L51 - it's required for the Netlify build.

**Comment by dholbach** [comment]
How do you feel about https://github.com/dholbach/website/tree/fix-129/adopters#readme - is this too complicated?

**Comment by dholbach** [comment]
I feel like this is getting somewhere. Things to be figured out still:

- [ ] Page description not working yet
- [ ] Size of logos
- [ ] No checking of paths
- [ ] carousel on main page

Do we already want to advertise the pages even if we haven't added companies yet?

File: adopters/flux-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  url: /adopters/flux-v2
+  description: >
+    Organisations below all are using [Flux v2](https://fluxcd.io) in production.
+      
+    We are happy and proud to have you all as part of our community! 💖
+  companies:
+    - name: Weaveworks
+      url: https://weave.works
+      logo: /img/logos/weaveworks.png
+    - name: Xenit SE
```
Comment (phillebaba) [inline adopters/flux-v2.yaml:12]: Could you changes this to just Xenit?

File: Makefile [hunk 2]
```diff
@@ -46,6 +46,10 @@ $(FONT_AWESOME_TARGET): ## Downloads the Docsy Font Awesome dependency.
 gen-content: ## Generates content from external sources.
 	hack/gen-content.sh
 	hack/import-flux2-assets.sh
+	command -v yq > /dev/null || test -x ./yq || \
+		curl -Lfs https://github.com/mikefarah/yq/releases/download/v4.7.1/yq_linux_amd64 -o yq && \
+		chmod a+x yq
```
Comment (dholbach) [inline Makefile:51]: This is not very nice. Can anyone help find a solution which works on the Netlify builds? (Ubuntu AFAIK.)

File: config.toml [hunk 3]
```diff
@@ -106,6 +106,18 @@ parent = "Project"
...
+parent = "Project"
+url = "/adopters/flux-v1"
+weight = 80
+
+[[menus.main]]
+name = "Flux v2 Adopters"
+parent = "Project"
+url = "/adopters/flux-v2"
+weight = 80
+
```
Comment (dholbach) [inline config.toml:120]: Maybe let's not advertise these pages yet?

**Comment by dholbach** [comment]
Let's move the main-page carousel to a separate PR.

**Comment by dholbach** [comment]
From now onwards, I'll need help making this pretty. :grin: :art: :sparkles:

File: adopters/flux-v1.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+adopters:
+  project: Flux v1
+  url: /adopters/flux-v1
```
Comment (dholbach) [inline adopters/flux-v1.yaml:3]: Is this the URL we want to use?

**Comment by dholbach** [comment]
Cards are the same size, I need help centering the logos now. :grin:

**Comment by dholbach** [comment]
Using card-deck with 5 entries per row. Fixes same size problem. Maybe good enough for now?

File: Makefile (hunk 2 above)
Comment (stefanprodan) [reply Makefile:51]: I see no issues with this, we do the same thing with GH Actions: https://github.com/fluxcd/flux2-kustomize-helm-example/blob/main/.github/actions/tools/entrypoint.sh

**Comment by scottrigby** [comment]
Minus adding logos for everyone, the adopters page looks good in general!

I do see something strange on a small viewport – on iPhone in portrait mode, cards are in a single line. In landscape mode, they alternate between rows of 3 and rows of 2. This could be iterated on though, as IMO it looks pretty good to start.

One thing – do we want to add some direct link to this adopters page from the main fluxcd.io landing page as part of this PR?

**Comment by dholbach** [comment]
Thanks for the feedback!

> Minus adding logos for everyone, the adopters page looks good in general!

Once it's up, we'll ask folks to add their logos.


> I do see something strange on a small viewport – on iPhone in portrait mode, cards are in a single line. In landscape mode, they alternate between rows of 3 and rows of 2. This could be iterated on though, as IMO it looks pretty good to start.

Yes, agreed - this is a known issue I'll need help with. Getting this into the shape it is was already a quite a bit of work (for me :innocent: )...


> One thing – do we want to add some direct link to this adopters page from the main fluxcd.io landing page as part of this PR?

Yes, if we feel it's good enough, I'm happy to add it. In the **Project** dropdown?

**Comment by dholbach** [comment]
I added it to the navbar under **Project**.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @dholbach 🏅

**Pull request merged**